microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -773,24 +782,24 @@ func getServiceDetailsFromManifests(filePath string) (string, int, error) {
 
 func isMetadataChanged(oldObject, newObject *metav1.ObjectMeta) bool {
 	if oldObject == nil || newObject == nil {
-		fmt.Printf("Metadata changes detected, old/new object is nil\n")
+		_log.Info("Metadata changes detected, old/new object is nil")
 		return oldObject != newObject
 	}
 	// only care limited changes
 	if oldObject.Name != newObject.Name {
-		fmt.Printf("Metadata changes detected, Name changed from %s to %s\n", oldObject.Name, newObject.Name)
+		_log.Info("Metadata.Name changes detected", "old", oldObject.Name, "new", newObject.Name)
 		return true
 	}
 	if oldObject.Namespace != newObject.Namespace {
-		fmt.Printf("Metadata changes detected, Namespace changed from %s to %s\n", oldObject.Namespace, newObject.Namespace)
+		_log.Info("Metadata.Namespace changes detected", "old", oldObject.Namespace, "new", newObject.Namespace)
 		return true
 	}
 	if !reflect.DeepEqual(oldObject.Labels, newObject.Labels) {
-		fmt.Printf("Metadata changes detected, Labels changed from %v to %v\n", oldObject.Labels, newObject.Labels)
+		_log.Info("Metadata.Labels changes detected", "old", oldObject.Labels, "new", newObject.Labels)
 		return true
 	}
 	if !reflect.DeepEqual(oldObject.DeletionTimestamp, newObject.DeletionTimestamp) {
-		fmt.Printf("Metadata changes detected, DeletionTimestamp changed from %v to %v\n", oldObject.DeletionTimestamp, newObject.DeletionTimestamp)
+		_log.Info("Metadata.DeletionTimestamp changes detected", "old", oldObject.DeletionTimestamp, "new", newObject.DeletionTimestamp)
```

Comment:
Shall this be on Debug level instead of Info level?

Reply:
from my view, this information is kind of useful for me, to know what condition trigger the reconcile, we don't want unexpected reconcile, if we hide this information to Debug, it would be less easier to find out such problem, espically for un-experienced developers.
Besides, there is no log.Debug, instead, it's log.V(2).Info

Reply:
We should add an option when controller start, to let user select the log level they want to set, eg: --logLevel debug
in such case, they can have much info in logs. 
It does not make sense to have most info output to Info level by default.

Reply:
> We should add an option when controller start, to let user select the log level they want to set, eg: `--logLevel debug` in such case, they can have much info in logs. It does not make sense to have most info output to Info level by default.

it's implemented